crates/rust-analyzer/src/global_state.rs

```diff
@@ -351,6 +351,10 @@ impl GlobalState {
 
                     crates.iter().any(|&krate| crate_graph[krate].is_proc_macro)
                 });
+            if self.proc_macro_changed && self.config.script_rebuild_on_save() {
+                self.fetch_build_data_queue
+                    .request_op(format!("proc-macro or build script source changed"), ())
+            }
```

Comment:
Hmm, so there is a problem with this for us here. Currently we set the `proc_macro_changed` flag whenever a change comes in for one of the sources, but those changes are not necessarily commited yet (saved), so with the current setup here we would ask for server rebuilds on every keystroke! We should instead move the op request to the `DidSaveTextDocument` handler (and then also reset the `proc_macro_changed` flag